### PR TITLE
Files/v6

### LINKED ISSUE #249: Handle slow stream inputs
+ Create a transactor per staging file
+ Split input byte stream by size _and_ idle time, i.e. if there is no data for 10min, end current stream
+ Write streams to temp files, emit them on inner stream end.
+ Upload these temp files removing them when upload succeded, retry on temp files too.

This way SF connection won't wait, but immediately upload stagin files, also much less memory will be consumed. The drawback is that it will take more time, IIUC the time is equal to maximum time one file is uploaded to staging. Which is not that bad especially for slow datasets.

### PR DESCRIPTION
**Pull request by cryogenian**
This includes #250 and fixes #249 

Snapshot `20.1.5-8981331` 

I can verify that atmost 2 temp files and 1 staged files are present simultaneously in this branch. 

Need verification from @beckyconning that this works as expected

### CODE REVIEW

**Comment by cryogenian** [comment]
BTW, I don't know why `Stream.resource` works now, I think that I made a mistake previously, but not sure where 😕

**Comment by cryogenian** [comment]
Becky said to go ahead with this approach (if I'm sure that this works, not much changed in comparison to #250, the files handling is the same)

**Comment by cryogenian** [comment]
Ah! Forgot to mention Hikari. In theory we could drop it, but I can't test that this will work correctly specifically for SAP cases, it seems to be a bit risky in such circumstances to remove connection pool.

File: core/src/main/scala/quasar/destination/snowflake/StageFile.scala [hunk 1]
```diff
@@ -63,143 +68,89 @@ object StageFile {
...
-                    elq.enqueue1(chunk.some) >>
-                    size.set(currentSize + chunk.size.toLong)
-                  }
-                } yield ()
-              }
-            Pull.eval(action) >> go(tail)
-          }
-        go(in).stream
-      }
+          case ExitCase.Error(e) =>
```
Comment (wemrysi) [inline core/src/main/scala/quasar/destination/snowflake/StageFile.scala:120]: Should be able replace the rest with a single wildcard match, no need to `raiseError` as this is a finalizer and doesn't affect the stream itself.

**Review by wemrysi — APPROVED** [review]

File: core/src/main/scala/quasar/destination/snowflake/StageFile.scala (hunk 1 above)
Comment (cryogenian) [reply core/src/main/scala/quasar/destination/snowflake/StageFile.scala:120]: :+1:

**Pull request merged**